pkg/model/piped_stat.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+const (
+	pipedStatsRetention    = 2 * time.Minute
+	pipedStatStaledTimeout = 24 * time.Hour
+)
+
+func UnmarshalPipedStat(data interface{}, ps *PipedStat) error {
+	value, okValue := data.([]byte)
+	if !okValue {
+		return errors.New("error value not a bulk of string value")
+	}
+	if err := json.Unmarshal(value, ps); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ps *PipedStat) IsStaled() bool {
+	return time.Since(time.Unix(ps.Timestamp, 0)) > pipedStatStaledTimeout
+}
+
+func IsConnectingPiped(ps *PipedStat) bool {
+	return time.Since(time.Unix(ps.Timestamp, 0)) < pipedStatsRetention
+}
```

Comment:
Or we can combine both `IsStaled`, `IsConnectingPiped`  into a single one

```
func (ps *PipedStat) IsStaled(d time.Duration) bool {
}

Reply:
I like this idea, lets me adopt that 👍

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2895/commits/a5e6f0cba955efc5fa6d78798c914278fb851177